File: tests/bench/loop_sum/LoopSum.java
// Same program as loop_sum.jai.
public class LoopSum {
    // BENCH_LEVEL picks how much work this benchmark does: easy a sixteenth of it,
    // medium a quarter, hard (the default, and anything unrecognised) all of it.
    static long scale() {
        String l = System.getenv("BENCH_LEVEL");
        if ("easy".equals(l)) return 16;
        if ("medium".equals(l)) return 4;
        return 1;
    }

    static final long SCALE = scale();
    static final long ITERATIONS = 50000000L / SCALE;

    public static void main(String[] args) {
        long total = 0;
        for (long i = 0; i < ITERATIONS; i++) {
            total += i % 7;
        }
        System.out.println(total);
    }
}

File: tests/bench/loop_sum/loop_sum.cpp
// Same program as loop_sum.jai. See tests/bench/README.md for why the C++ and
// Java rows are here and what they are not.
#include <cstdint>
#include <cstdio>

#include <cstdlib>
#include <cstring>

// BENCH_LEVEL picks how much work this benchmark does: easy a sixteenth of it,
// medium a quarter, hard (the default, and anything unrecognised) all of it.
static long bench_scale() {
    const char *l = std::getenv("BENCH_LEVEL");
    if (l && std::strcmp(l, "easy") == 0) return 16;
    if (l && std::strcmp(l, "medium") == 0) return 4;
    return 1;
}
static const long SCALE = bench_scale();
static const int64_t ITERS = 50000000 / SCALE;

int main() {
    int64_t total = 0;
    for (int64_t i = 0; i < ITERS; i++) {
        total += i % 7;
    }
    std::printf("%lld\n", (long long)total);
    return 0;
}
